source/main/gfx/GfxActor.cpp

```diff
@@ -1202,3 +1333,992 @@ void RoR::GfxActor::CalculateDriverPos(Ogre::Vector3& out_pos, Ogre::Quaternion&
+        if (prop.beacon_light_rotation_angle[0] > 1.0)
+        {
+            prop.beacon_light_rotation_angle[0] = 0.0;
+            visible = true;
+        }
+        visible = visible && enableAll;
+        prop.beacon_light[0]->setVisible(visible);
+        prop.beacon_flare_billboard_scene_node[0]->setVisible(visible);
+    }
+    if (prop.beacontype == 'R' || prop.beacontype == 'L')
+    {
+        Vector3 mposition = nodes[prop.noderef].AbsPosition + prop.offsetx * (nodes[prop.nodex].AbsPosition - nodes[prop.noderef].AbsPosition) + prop.offsety * (nodes[prop.nodey].AbsPosition - nodes[prop.noderef].AbsPosition);
+        //billboard
+        Vector3 vdir = mposition - gEnv->mainCamera->getPosition();
+        float vlen = vdir.length();
+        if (vlen > 100.0)
+        {
+            prop.beacon_flare_billboard_scene_node[0]->setVisible(false);
+            return;
+        }
+        //normalize
+        vdir = vdir / vlen;
+        prop.beacon_flare_billboard_scene_node[0]->setPosition(mposition - vdir * 0.1);
+    }
+    if (prop.beacontype == 'w')
+    {
+        Vector3 mposition = nodes[prop.noderef].AbsPosition + prop.offsetx * (nodes[prop.nodex].AbsPosition - nodes[prop.noderef].AbsPosition) + prop.offsety * (nodes[prop.nodey].AbsPosition - nodes[prop.noderef].AbsPosition);
+        prop.beacon_light[0]->setPosition(mposition);
+        prop.beacon_light_rotation_angle[0] += dt * prop.beacon_light_rotation_rate[0];//rotate baby!
+        //billboard
+        Vector3 vdir = mposition - gEnv->mainCamera->getPosition();
+        float vlen = vdir.length();
+        if (vlen > 100.0)
+        {
+            prop.beacon_flare_billboard_scene_node[0]->setVisible(false);
+            return;
+        }
+        //normalize
+        vdir = vdir / vlen;
+        prop.beacon_flare_billboard_scene_node[0]->setPosition(mposition - vdir * 0.1);
+        bool visible = false;
+        if (prop.beacon_light_rotation_angle[0] > 1.0)
+        {
+            prop.beacon_light_rotation_angle[0] = 0.0;
+            visible = true;
+        }
+        visible = visible && enableAll;
+        prop.beacon_light[0]->setVisible(visible);
+        prop.beacon_flare_billboard_scene_node[0]->setVisible(visible);
+    }
+}
+
+void RoR::GfxActor::UpdateProps(float dt, bool is_player_actor)
+{
+    using namespace Ogre;
+
+    if (m_simbuf.simbuf_beaconlight_active != m_beaconlight_active)
+    {
+        m_beaconlight_active = m_simbuf.simbuf_beaconlight_active;
+        this->SetBeaconsEnabled(m_beaconlight_active);   
+    }
+
+    NodeData* nodes = this->GetSimNodeBuffer();
+
+    for (prop_t& prop: m_props)
+    {
+        if (!prop.scene_node)
+            continue;
+
+        // -- quick ugly port from `Actor::NotifyCameraChanged()`~ 06/2018
+        const bool mo_visible = (prop.cameramode == -2 || prop.cameramode == m_simbuf.simbuf_cur_cinecam);
+        if (prop.mo != nullptr)
+        {
+            prop.mo->setVisible(mo_visible);
```

Comment:
This used to be `setMeshEnabled`.

Reply:
Perhaps a mistake, I really don't know, the whole `MeshObject` is a piece of ****, see this branch I just made: https://github.com/only-a-ptr/rigs-of-rods/commits/MeshObject-cleanup